pkg/controller/apimanager/apimanager_controller.go

```diff
@@ -0,0 +1,580 @@
+	c, err := controller.New("apimanager-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource APIManager
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.APIManager{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to DeploymentConfigs to update deployment status
+	ownerHandler := &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &appsv1alpha1.APIManager{},
+	}
+	err = c.Watch(&source.Kind{Type: &appsv1.DeploymentConfig{}}, ownerHandler)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileAPIManager implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAPIManager{}
+
+// ReconcileAPIManager reconciles a APIManager object
+type ReconcileAPIManager struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client          client.Client
+	scheme          *runtime.Scheme
+	reqLogger       logr.Logger
+	apiClientReader client.Reader
+}
+
+// Reconcile reads that state of the cluster for a APIManager object and makes changes based on the state read
+// and what is in the APIManager.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileAPIManager) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.reqLogger = log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	r.reqLogger.Info("Reconciling APIManager")
+
+	// Fetch the APIManager instance
+	instance := &appsv1alpha1.APIManager{}
+
+	r.reqLogger.Info("Trying to get APIManager resource")
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.reqLogger.Info("APIManager Resource not found. Ignoring since object must have been deleted")
+			return reconcile.Result{}, nil
+		}
+		r.reqLogger.Error(err, "APIManager Resource cannot be created. Requeuing request...")
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	r.reqLogger.Info("Successfully retreived APIManager resource")
+
+	r.reqLogger.Info("Setting defaults for APIManager resource")
+	changed, err := instance.SetDefaults() // TODO check where to put this
+	if err != nil {
+		// Error setting defaults - Stop reconciliation
+		return reconcile.Result{}, nil
```

Comment:
Don't you want to return the error to try again, in case of intermittent errors?
`return reconcile.Result{}, err`

Reply:
iep, just detected this yesterday when reviewing the reconciler code :+1: will fix it